Revisit Session Management docs

marcusdacoregio opened the pull request:
Closes gh-12519

jzheaux reviewed (changes requested):
Nice, @marcusdacoregio! I've left my feedback inline.

marcusdacoregio commented on docs/modules/ROOT/pages/servlet/authentication/session-management.adoc:
```diff
@@ -16,53 +13,297 @@ This can be done by using the {security-api-url}org/springframework/security/web
+If you are not sure what `securityContextHolderStrategy` is in the above example, you can read more about it in the <<use-securitycontextholderstrategy, Using `SecurityContextStrategy` section>>.
+
+[[stateless-authentication]]
+=== Configuring persistence for Stateless Authentication
```
I think I'll remove the `RequestCache` section and a ticket can be created to add some information about that.

marcusdacoregio commented:
Thanks, @jzheaux, I've addressed your suggestions and updated the PR. I also reverted [this commit](https://github.com/spring-projects/spring-security/pull/12601/commits/19ccbbaec598c9d857440b7b4e54c84eab6eb673) based on @rwinch feedback.

jzheaux reviewed (approved):
Such a huge improvement, @marcusdacoregio. Thanks for your efforts.
